eth/peerset.go

```diff
@@ -129,7 +131,17 @@ func (ps *peerSet) waitSnapExtension(peer *eth.Peer) (*snap.Peer, error) {
 	ps.snapWait[id] = wait
 	ps.lock.Unlock()
 
-	return <-wait, nil
+	for {
```

Comment:
Why the `for` ? I think the standalone `select` is sufficient -- both cases exit with a `return`, so there's no looping

Reply:
You are right. It was modified on the basis of handling the timeout channel, so there was an oversight here.